man/man3/ConfigRam.3
.TH "ConfigRam" 3 "Thu Aug 23 2012" "Version 1.0" "FS-UAE Gui for Linux OS" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ConfigRam \- 
.PP
This class is responsible for the Ram Configuration management\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <configram\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConfigRam\fP ()"
.br
.ti -1c
.RI "string \fBgetChipMemoryConfigString\fP ()"
.br
.RI "\fIRepresents the string of Chip memory to be written in Config File\&. \fP"
.ti -1c
.RI "string \fBgetChipMemoryString\fP ()"
.br
.RI "\fIRepresent the effective value of Chip memory\&. \fP"
.ti -1c
.RI "string \fBgetSlowMemoryConfigString\fP ()"
.br
.RI "\fIRepresents the string of Slow memory to be written in Config File\&. \fP"
.ti -1c
.RI "string \fBgetSlowMemoryString\fP ()"
.br
.RI "\fIRepresent the effective value of Slow memory\&. \fP"
.ti -1c
.RI "string \fBgetFastMemoryConfigString\fP ()"
.br
.RI "\fIRepresents the string of Zorro II Fast memory to be written in Config File\&. \fP"
.ti -1c
.RI "string \fBgetFastMemoryString\fP ()"
.br
.RI "\fIRepresent the effective value of Zorro II Fast memory\&. \fP"
.ti -1c
.RI "string \fBgetZorro3ConfigString\fP ()"
.br
.RI "\fIRepresents the string of Zorro III Fast memory to be written in Config File\&. \fP"
.ti -1c
.RI "string \fBgetZorro3String\fP ()"
.br
.RI "\fIRepresent the effective value of Zorro III Fast memory\&. \fP"
.ti -1c
.RI "int \fBsetParameter\fP (string parameter, string value)"
.br
.ti -1c
.RI "void \fBsetToDefaultConfiguration\fP ()"
.br
.ti -1c
.RI "bool \fBhasParameter\fP (string parameterName)"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "string \fBchip_memory\fP"
.br
.RI "\fIOverride amount of chip memory, specified in KB\&. Must be a multiple of 512\&. \fP"
.ti -1c
.RI "string \fBslow_memory\fP"
.br
.RI "\fIOverride the amount of “slow” memory, specified in KB\&. Must be a multiple of 256\&. \fP"
.ti -1c
.RI "string \fBfast_memory\fP"
.br
.RI "\fIOverride the amount of Zorro II Fast memory, specified in KB\&. Must be a multiple of 1024\&. \fP"
.ti -1c
.RI "string \fBzorro_iii_memory\fP"
.br
.RI "\fIOverride the amount of Zorro III Fast memory, specified in KB\&. Must be a multiple of 1024\&.
.br
 Requires a processor with 32­bit address bus, (use for example the A1200/020 model)\&. \fP"
.in -1c
.SH "Detailed Description"
.PP 
This class is responsible for the Ram Configuration management\&. 

The default value of each type of ram depends on the selected \fBAmiga\fP model\&. 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "\fBConfigRam::ConfigRam\fP ()"
.SH "Member Function Documentation"
.PP 
.SS "string \fBConfigRam::getChipMemoryConfigString\fP ()"
.PP
Represents the string of Chip memory to be written in Config File\&. If Chip memory is the default value, it returns an empty string\&.
.br
 chip_memory = 1024 
.SS "string \fBConfigRam::getChipMemoryString\fP ()"
.PP
Represent the effective value of Chip memory\&. 
.SS "string \fBConfigRam::getFastMemoryConfigString\fP ()"
.PP
Represents the string of Zorro II Fast memory to be written in Config File\&. If Fast memory is the default value, it returns an empty string\&.
.br
 fast_memory = 4096 
.SS "string \fBConfigRam::getFastMemoryString\fP ()"
.PP
Represent the effective value of Zorro II Fast memory\&. 
.SS "string \fBConfigRam::getSlowMemoryConfigString\fP ()"
.PP
Represents the string of Slow memory to be written in Config File\&. If Slow memory is the default value, it returns an empty string\&.
.br
 slow_memory = 512 
.SS "string \fBConfigRam::getSlowMemoryString\fP ()"
.PP
Represent the effective value of Slow memory\&. 
.SS "string \fBConfigRam::getZorro3ConfigString\fP ()"
.PP
Represents the string of Zorro III Fast memory to be written in Config File\&. If Zorro III memory is the default value, it returns an empty string\&.
.br
 zorro_iii_memory = 65536 
.SS "string \fBConfigRam::getZorro3String\fP ()"
.PP
Represent the effective value of Zorro III Fast memory\&. 
.SS "bool \fBConfigRam::hasParameter\fP (stringparameterName)"
.SS "int \fBConfigRam::setParameter\fP (stringparameter, stringvalue)"
.SS "void \fBConfigRam::setToDefaultConfiguration\fP ()"
.SH "Member Data Documentation"
.PP 
.SS "string \fBConfigRam::chip_memory\fP\fC [private]\fP"
.PP
Override amount of chip memory, specified in KB\&. Must be a multiple of 512\&. 
.SS "string \fBConfigRam::fast_memory\fP\fC [private]\fP"
.PP
Override the amount of Zorro II Fast memory, specified in KB\&. Must be a multiple of 1024\&. 
.SS "string \fBConfigRam::slow_memory\fP\fC [private]\fP"
.PP
Override the amount of “slow” memory, specified in KB\&. Must be a multiple of 256\&. 
.SS "string \fBConfigRam::zorro_iii_memory\fP\fC [private]\fP"
.PP
Override the amount of Zorro III Fast memory, specified in KB\&. Must be a multiple of 1024\&.
.br
 Requires a processor with 32­bit address bus, (use for example the A1200/020 model)\&. 

.SH "Author"
.PP 
Generated automatically by Doxygen for FS-UAE Gui for Linux OS from the source code\&.


configram.cpp
/*! \file configram.cpp
 configram.cpp

 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 Written by Riva Alessandro Mario, alessandromario.riva@gmail.com
 and Sanvito Davide, dsanvito90@gmail.com. */
#include "configram.h"

const string DEFAULTCHIPMEMORY = "512";
const string DEFAULTSLOWMEMORY = "NONE";
const string DEFAULTFASTMEMORY = "NONE";
const string DEFAULTZORROIIIMEMORY = "NONE";

ConfigRam::ConfigRam()
{
    setToDefaultConfiguration();
}

string ConfigRam::getChipMemoryConfigString()
{
    if (chip_memory.compare(DEFAULTCHIPMEMORY)==0){return "";}
    else {return "chip_memory = " + chip_memory;}
}

string ConfigRam::getChipMemoryString()
{
    return chip_memory;
}
//////////////////////////////////////////////////////////////////////////////////////////
string ConfigRam::getSlowMemoryConfigString()
{
    if (slow_memory.compare(DEFAULTSLOWMEMORY)==0){return "";}
    else {return "slow_memory = " + slow_memory;}
}

string ConfigRam::getSlowMemoryString()
{
    return slow_memory;
}
//////////////////////////////////////////////////////////////////////////////////////////
string ConfigRam::getFastMemoryConfigString()
{
    if ((fast_memory.compare(DEFAULTFASTMEMORY)==0)){return "";} //||(chip_memory.compare("4096")==0)||(chip_memory.compare("8192")==0)
    else {return "fast_memory = " + fast_memory;}
}

string ConfigRam::getFastMemoryString()
{
    return fast_memory;
}
//////////////////////////////////////////////////////////////////////////////////////////
string ConfigRam::getZorro3ConfigString()
{
    if (zorro_iii_memory.compare(DEFAULTZORROIIIMEMORY)==0){return "";}
    else {return "zorro_iii_memory = " + zorro_iii_memory;}
}

string ConfigRam::getZorro3String()
{
    return zorro_iii_memory;
}
//////////////////////////////////////////////////////////////////////////////////////////

int ConfigRam::setParameter(string parameter, string value)
{
    //se mi arriva un parametro che non so/devo gestire io non faccio nulla
    //se mi arriva un mio parametro verifico sia valido, se no setto il DEFAULT
    //in questo modo se qualcuno salva la config, manomette (o modifica erroneamente) il file, io garantisco sia una configurazione valida
    if(parameter.compare("chip_memory")==0){
        if ((value.compare("512")==0)||(value.compare("1024")==0)||(value.compare("2048")==0)||(value.compare("4096")==0)||(value.compare("8192")==0)){
            chip_memory=value;}
        else{
            chip_memory=DEFAULTCHIPMEMORY;
            return -1;
        }
    } else if(parameter.compare("slow_memory")==0){
        if ((value.compare("NONE")==0)||(value.compare("512")==0)||(value.compare("1024")==0)||(value.compare("1792")==0)){
            slow_memory=value;}
        else{
            slow_memory=DEFAULTSLOWMEMORY;
            return -1;
        }
    } else if(parameter.compare("fast_memory")==0){
        if ((value.compare("NONE")==0)||(value.compare("1024")==0)||(value.compare("2048")==0)||(value.compare("4096")==0)||(value.compare("8192")==0)){
            fast_memory=value;}
        else{
            fast_memory=DEFAULTFASTMEMORY;
            return -1;
        }
    } else if(parameter.compare("zorro_iii_memory")==0){
        if ((value.compare("NONE")==0)||(value.compare("1024")==0)||(value.compare("2048")==0)||(value.compare("4096")==0)||(value.compare("8192")==0)||(value.compare("16384")==0)||(value.compare("32768")==0)||(value.compare("65536")==0)||(value.compare("131072")==0)||(value.compare("262144")==0)){
            zorro_iii_memory=value;}
        else{
            zorro_iii_memory=DEFAULTZORROIIIMEMORY;
            return -1;
        }
    }
    return 0;
}

void ConfigRam::setToDefaultConfiguration()
{
    chip_memory=DEFAULTCHIPMEMORY;
    slow_memory=DEFAULTSLOWMEMORY;
    fast_memory=DEFAULTFASTMEMORY;
    zorro_iii_memory=DEFAULTZORROIIIMEMORY;

}

bool ConfigRam::hasParameter(string parameterName)
{
    return (parameterName.compare("chip_memory")==0)||(parameterName.compare("slow_memory")==0)||(parameterName.compare("fast_memory")==0)||(parameterName.compare("zorro_iii_memory")==0);
}
